app/models/goal.py

```diff
@@ -2,5 +2,26 @@
 from app import db
 
 
+
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement = True)
+    title = db.Column(db.String, nullable=False)
+    tasks = db.relationship("Task", backref="goal", lazy=True)
+
+    def is_complete(self):
+        if self.completed_at is None:
+            return False
+        else:
+            return True
+
+    def to_dict(self):
+        return {
+            "id": self.goal_id, 
+            "title": self.title
+            }   
+
+    def to_json(self):
+        return {
+            "id": self.goal_id,
+            "title": self.title,
+        }
```

Comment:
These two are redundant (as indicated by that fact that the code within them looks the exact same). 
It's great to have both of them but having two functions that do the same thing with different names is not recommended.